File: app/src/main/java/com/gameautopilot/app/util/Logger.kt
package com.gameautopilot.app.util

import android.util.Log

object Logger {
    private const val TAG = "Autopilot"

    fun d(msg: String) = Log.d(TAG, msg)
    fun i(msg: String) = Log.i(TAG, msg)
    fun w(msg: String, t: Throwable? = null) = Log.w(TAG, msg, t)
    fun e(msg: String, t: Throwable? = null) = Log.e(TAG, msg, t)
}

File: mac/Sources/GameAutopilotMac/Util/Logger.swift
import os

/// Thin wrapper over unified logging. Mirrors util/Logger.kt's shape.
/// View live output with: log stream --predicate 'subsystem == "com.gameautopilot.mac"'
enum Logger {
    private static let log = os.Logger(subsystem: "com.gameautopilot.mac", category: "Autopilot")

    static func d(_ msg: String) { log.debug("\(msg, privacy: .public)") }
    static func i(_ msg: String) { log.info("\(msg, privacy: .public)") }
    static func w(_ msg: String) { log.warning("\(msg, privacy: .public)") }
    static func e(_ msg: String) { log.error("\(msg, privacy: .public)") }
}
